src/vs/base/browser/ui/list/rangeMap.ts

```diff
@@ -91,6 +91,21 @@ export class RangeMap {
 
 	private groups: IRangedGroup[] = [];
 	private _size = 0;
+	private _topPadding = 0;
+
+	get topPadding() {
+		return this._topPadding;
+	}
+
+	set topPadding(topPadding: number) {
+		this._topPadding = topPadding;
+		this._size = this._topPadding + this.groups.reduce((t, g) => t + (g.size * (g.range.end - g.range.start)), 0);
```

Comment:
Can't we do better here? Size should only change by `topPadding - this._topPadding`, no need to iterate over items.